### PR TITLE
Known addresses v3

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1430

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/transactions/details/view/LabeledNamedAddressItem.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import io.gnosis.safe.databinding.ViewLabeledNamedAddressItemBinding
+import pm.gnosis.model.Solidity
+
+
+class LabeledNamedAddressItem @JvmOverloads constructor(
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/transactions/details/view/LabeledNamedAddressItem.kt:11]: Shouldn't we be able to use this in a lot more places?

Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/transactions/details/view/LabeledNamedAddressItem.kt:11]: please elaborate on this

File: app/src/test/java/io/gnosis/safe/ui/settings/owner/intro/OwnerGenerateViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import io.gnosis.safe.utils.MnemonicKeyAndAddressDerivator
+import io.mockk.*
+import org.junit.Rule
+import org.junit.Test
+import pm.gnosis.mnemonic.Bip39
+import pm.gnosis.mnemonic.Bip39.Companion.MIN_ENTROPY_BITS
+import pm.gnosis.model.Solidity
+import java.math.BigInteger
+
+class OwnerGenerateViewModelTest {
```
Comment (biafra23) [inline app/src/test/java/io/gnosis/safe/ui/settings/owner/intro/OwnerGenerateViewModelTest.kt:15]: This is unrelated to the rest of the PR? I don't mind, if it is.

File: data/src/main/java/io/gnosis/data/models/transaction/TransactionDetails.kt [hunk 3]
```diff
@@ -37,7 +39,11 @@ data class TxData(
     @Json(name = "value")
     val value: BigInteger?,
     @Json(name = "operation")
-    val operation: Operation
+    val operation: Operation,
+
+    //FIXME: suboptimal backend response structure
```
Comment (biafra23) [inline data/src/main/java/io/gnosis/data/models/transaction/TransactionDetails.kt:44]: This will be different with the new endpoint, right?

**Review by biafra23 — APPROVED** [review]

File: app/src/test/java/io/gnosis/safe/ui/settings/owner/intro/OwnerGenerateViewModelTest.kt (hunk 2 above)
Comment (elgatovital) [reply app/src/test/java/io/gnosis/safe/ui/settings/owner/intro/OwnerGenerateViewModelTest.kt:15]: it is here to increase test coverage

File: data/src/main/java/io/gnosis/data/models/transaction/TransactionDetails.kt (hunk 3 above)
Comment (elgatovital) [reply data/src/main/java/io/gnosis/data/models/transaction/TransactionDetails.kt:44]: I hope so
